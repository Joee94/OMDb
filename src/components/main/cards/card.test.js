import Card from './card';

describe('Card', () => {
	describe('poster exists', () => {
		const movie = { Title: 'star wars', Poster: 'star wars poster', Year: '1999', imdbRating: '5/5' };
		let returnValue;
		beforeEach(() => {
			returnValue = Card({ movie });
		});
		it('should create the Card component', () => {
			expect(returnValue).toMatchSnapshot();
		});
	});
	describe('poster does not exist', () => {
		const movie = { Title: 'star wars', Poster: 'N/A', Year: '1999', imdbRating: '5/5' };
		let returnValue;
		beforeEach(() => {
			returnValue = Card({ movie });
		});
		it('should create the Card component', () => {
			expect(returnValue).toMatchSnapshot();
		});
	});
});
