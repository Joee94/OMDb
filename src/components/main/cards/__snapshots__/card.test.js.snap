// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card poster does not exist should create the Card component 1`] = `
<div
  aria-label="star wars. Released 1999, Rated 5/5 on IMDB"
  aria-required="true"
  className="card"
  tabIndex="0"
>
  <img
    alt="Missing poster for star wars"
    onError={[Function]}
    src="noImage.png"
  />
  <div
    className="text"
  >
    <h1>
      star wars
    </h1>
    <span>
      <h2
        className="label"
      >
        Year:
      </h2>
       
      <h2
        className="data"
      >
        1999
      </h2>
    </span>
    <span>
      <h2
        className="label"
      >
        Rating:
      </h2>
       
      <h2
        className="data"
      >
        5/5
      </h2>
    </span>
  </div>
</div>
`;

exports[`Card poster exists should create the Card component 1`] = `
<div
  aria-label="star wars. Released 1999, Rated 5/5 on IMDB"
  aria-required="true"
  className="card"
  tabIndex="0"
>
  <img
    alt="Poster for star wars"
    onError={[Function]}
    src="star wars poster"
  />
  <div
    className="text"
  >
    <h1>
      star wars
    </h1>
    <span>
      <h2
        className="label"
      >
        Year:
      </h2>
       
      <h2
        className="data"
      >
        1999
      </h2>
    </span>
    <span>
      <h2
        className="label"
      >
        Rating:
      </h2>
       
      <h2
        className="data"
      >
        5/5
      </h2>
    </span>
  </div>
</div>
`;
